cypress/e2e/customers.cy.js

```diff
@@ -0,0 +1,142 @@
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(response.body).to.have.property('customers');
+      expect(response.body.customers).to.be.an('array');
+      expect(response.body.customers).to.have.lengthOf(20);
+      //  expect(response.body.pageInfo).to.have.property('currentPage', 1);
+      expect(response.body.pageInfo).to.have.property('totalPages', 3);
+      expect(response.body.pageInfo).to.have.property('totalCustomers', 50);
+    });
+  });
+
+  it('Should handle invalid page numbers gracefully', () => {
+    cy.request({
+      method: 'GET',
+      url: 'http://localhost:3001/customers?page=-1&limit=10&size=All',
+      failOnStatusCode: false
+    }).then((response) => {
+      expect(response.status).to.eq(400); // Assuming the API returns 400 for invalid page numbers
+      expect(response.body).to.have.property('error');
+      expect(response.body.error).to.contain('Invalid page or limit. Both must be positive numbers');
+    });
+  });
+/*
+  it('Check individual properties', () => {
+    cy.request({
+      method: 'GET',
+      url: 'http://localhost:3001/customers?page=1&limit=10&size=All',
+    }).then((response) => {
+      // Check response status
+      expect(response.status).to.eq(200);
+
+      // Check that the response has 'customers' and 'pageInfo'
+      expect(response.body).to.have.property('customers').that.is.an('array').with.length(10);
+      expect(response.body).to.have.property('pageInfo');
+
+      // Check each customer object in the array
+      response.body.customers.forEach((customer, index) => {
+        expect(customer).to.have.property('id').that.is.a('number');
+        expect(customer).to.have.property('name').that.is.a('string');
+        expect(customer).to.have.property('employees').that.is.a('number');
+
+        if (customer.contactInfo) {
+          expect(customer.contactInfo).to.have.property('name').that.is.a('string');
+          expect(customer.contactInfo).to.have.property('email').that.is.a('string');
+        } else {
+          expect(customer).to.have.property('contactInfo', null);
+        }
+
+       // expect(customer.address).to.have.property('street').that.is.a('string');
+        expect(customer.address).to.have.property('city').that.is.a('string');
+        expect(customer.address).to.have.property('state').that.is.a('string');
+        expect(customer.address).to.have.property('zipCode').that.is.a('string');
+        expect(customer.address).to.have.property('country').that.is.a('string');
+
+        expect(customer).to.have.property('size').that.is.a('string');
+      });
+
+      // Additional checks on the 'pageInfo' object
+      expect(response.body.pageInfo).to.have.property('currentPage').that.is.a('string');
+      expect(response.body.pageInfo).to.have.property('totalPages').that.is.a('number');
+      expect(response.body.pageInfo).to.have.property('totalCustomers').that.is.a('number');
+    });
+  });*/
+});
```

Comment:
```suggestion
});

```

Every file should end with an empty line.

By doing so, tools like GitHub don't show the icon of "No newline at end of file".